### PR TITLE
Add changelog note for Go extraction reconciliation

### PR DESCRIPTION
**Pull request by angelapwen**
### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## [UNRELEASED]
 
-No user facing changes.
+- We will soon be rolling out a feature of the CodeQL Action that treats Go as a compiled language for the purposes of code scanning, rather than as a scanned language. Once the feature is fully rolled out, additional documentation will be released. If you see issues with your existing workflows that analyze Go, please [file an issue](https://github.com/github/codeql-core/issues) or contact support.
```
Comment (henrymercer) [inline CHANGELOG.md:5]:
```suggestion
- We will soon be rolling out a feature of the CodeQL Action that changes the way that Go code is analyzed to be more consistent with other compiled languages. You do not need to change your code scanning workflows. If you encounter any problems, please [file an issue](https://github.com/github/codeql-action/issues) or open a private ticket with GitHub Support and request an escalation to engineering.
```

**Review by henrymercer — APPROVED** [review]
Some small recommendations

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 ## [UNRELEASED]
 
-No user facing changes.
+- We will soon be rolling out a feature of the CodeQL Action that changes the way that Go code is analyzed to be more consistent with other compiled languages. You do not need to change your code scanning workflows. If you encounter any problems, please [file an issue](https://github.com/github/codeql-action/issues) or open a private ticket with GitHub Support and request an escalation to engineering.
```
Comment (aeisenberg) [inline CHANGELOG.md:5]: "soon" is a little fuzzy here. _I_ know why you are using it (gradually enabling the feature through server side flags), but for users that's not apparent. How about just removing "soon"?

Also, it's ambiguous what the change is. Maybe that's ok, but is there a little  bit more of a hint we can give users about what is going on?

Does this sound better?

```suggestion
- We are rolling out a change to the way that Go code is analyzed to be more consistent with other compiled languages. You do not need to alter your code scanning workflows. If you encounter any problems, please [file an issue](https://github.com/github/codeql-action/issues) or open a private ticket with GitHub Support and request an escalation to engineering.
```

Comment (henrymercer) [inline CHANGELOG.md:5]:
```suggestion
- We will be rolling out a feature of the CodeQL Action in October 2022 that changes the way that Go code is analyzed to be more consistent with other compiled languages like C/C++, C#, and Java. You do not need to change your code scanning workflows. If you encounter any problems, please [file an issue](https://github.com/github/codeql-action/issues) or open a private ticket with GitHub Support and request an escalation to engineering.
```

**Review by henrymercer — DISMISSED** [review]
Further tweaks from Slack

File: CHANGELOG.md (hunk 2 above)
Comment (angelapwen) [reply CHANGELOG.md]: Oops, I committed Andrew's suggestion and can't auto-commit this one anymore from the UI. Will push it up locally.

**Comment by angelapwen** [comment]
Need another non-stale ✅ from @henrymercer or @aeisenberg 🙏

**Review by aeisenberg — APPROVED** [review]

**Pull request merged**